lib/python/fidasim/preprocessing.py

```diff
@@ -1247,6 +1264,10 @@ def write_namelist(filename, inputs):
         f.write("origin(2) = {:f}     !! V value [cm]\n".format(inputs['origin'][1]))
         f.write("origin(3) = {:f}     !! W value [cm]\n\n".format(inputs['origin'][2]))
 
+        f.write("!! Adaptive time step settings\n")
+        f.write("cell_split = {:d}     !! 0: split off, 1: user-input split, 2: tolerance split\n".format(inputs['cell_split']))
+        f.write("tol = {:f}     !! Tolerance for adaptive step size\n\n".format(inputs['tol']))
+
```

Comment:
I would have the meaning of `tol` change according to the value of `cell_split`. For example if `cell_split = 1` then tol is a step size in cm. If `cell_split = 2` then it is a tolerance. However this also mixes up the definition for `tol` so you could also have `tol` be the fraction of a cell crossing instead of a literal distance.